app/models/like.rb

```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: :likes_counter
+
+  def update_counters
+    update(likes_counter: likes.count)
+  end
```

Comment:
- [ ] As in the previous comment. I think it would be better if you update your ` update_counters` method of **likes** to match the required result. According to this **[article](https://www.geeksforgeeks.org/ruby-string-count-method/)** the `increment` method will be good than the `count` method. As you are supposed to count the newly added **likes**. **[Here](https://stackoverflow.com/questions/11499110/increment-vs-1)** is an example of how to use it.

-- [ **OPTIONAL** ] I can also suggest When implementing both the `update_likes_counter` and `update_comments_counter`  methods, you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database.
```ruby
# Example
  ...
  after_save :update_post_counter

  private

  def update_post_counter
    # Your code here...
  end
```

Reply:
Changes made